packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, Prop, h, Listen} from '@stencil/core';
+import {buildInteractiveResult, Result, SmartSnippet} from '@coveo/headless';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {ResultContextEvent} from '../result-template-components/result-template-decorators';
+
+@Component({
+  tag: 'atomic-smart-snippet-source',
+  shadow: false,
+})
+export class AtomicSmartSnippetSource implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Prop() source!: Result;
+  @Prop() smartSnippet: SmartSnippet | undefined;
```

Comment:
I think we should replace the `smartSnippet` prop with either three callback parameters, or preferably three event emitters.

You can copy the event emitter pattern from `atomic-modal` (which emits `animationEnded`) and `atomic-refine-modal` (which listens to it through `onAnimationEnded`).

Reply:
One thing I just noticed I forgot: the `@Event()` should have `bubbles: false` like so: `@Event({bubbles: false})`